modules/core/src/main/java/org/apache/synapse/api/API.java

```diff
@@ -276,20 +280,14 @@ public boolean canProcess(MessageContext synCtx) {
                 return false;
             }
         } else {
-            String path = ApiUtils.getFullRequestPath(synCtx);
-            if (null == synCtx.getProperty(RESTConstants.IS_PROMETHEUS_ENGAGED) &&
-                    (!ApiUtils.matchApiPath(path, context))) {
-                auditDebug("API context: " + context + " does not match request URI: " + path);
-                return false;
-            }
-
-            if(!versionStrategy.isMatchingVersion(synCtx)){
+            if (!ApiUtils.identifyApi(this, synCtx)) {
                 return false;
+            }else{
+                ApiUtils.getFullRequestPath(synCtx);
```

Comment:
can we simply move this out of the else block?

Reply:
Fixed